flink-table/flink-sql-parser-hive/src/main/java/org/apache/flink/sql/parser/hive/ddl/SqlCreateHiveTable.java

```diff
@@ -0,0 +1,467 @@
+			SqlTableColumn partCol = (SqlTableColumn) node;
+			res.add(partCol.getName());
+		}
+		return res;
+	}
+
+	/**
+	 * Creation context for a Hive table.
+	 */
+	public static class HiveTableCreationContext extends TableCreationContext {
+		public SqlIdentifier pkName = null;
+		public Byte pkTrait = null;
+		public SqlIdentifier ukName = null;
+		public List<Byte> notNullTraits = null;
+	}
+
+	/**
+	 * To represent STORED AS in CREATE TABLE DDL.
+	 */
+	public static class HiveTableStoredAs {
+
+		public static final String STORED_AS_FILE_FORMAT = "hive.stored.as.file.format";
+		public static final String STORED_AS_INPUT_FORMAT = "hive.stored.as.input.format";
+		public static final String STORED_AS_OUTPUT_FORMAT = "hive.stored.as.output.format";
+
+		private final SqlParserPos pos;
+		private final SqlIdentifier fileFormat;
+		private final SqlCharStringLiteral intputFormat;
+		private final SqlCharStringLiteral outputFormat;
+
+		private HiveTableStoredAs(SqlParserPos pos, SqlIdentifier fileFormat, SqlCharStringLiteral intputFormat,
+				SqlCharStringLiteral outputFormat) throws ParseException {
+			this.pos = pos;
+			this.fileFormat = fileFormat;
+			this.intputFormat = intputFormat;
+			this.outputFormat = outputFormat;
+			validate();
+		}
+
+		private void validate() throws ParseException {
+			if (fileFormat != null) {
+				if (intputFormat != null || outputFormat != null) {
+					throw new ParseException("Both file format and input/output format are specified");
+				}
+			} else {
+				if (intputFormat == null || outputFormat == null) {
+					throw new ParseException("Neither file format nor input/output format is specified");
+				}
+			}
+		}
+
+		public SqlNodeList toPropList() {
+			SqlNodeList res = new SqlNodeList(pos);
+			if (fileFormat != null) {
+				res.add(HiveDDLUtils.toTableOption(STORED_AS_FILE_FORMAT, fileFormat.getSimple(), fileFormat.getParserPosition()));
+			} else {
+				res.add(HiveDDLUtils.toTableOption(STORED_AS_INPUT_FORMAT, intputFormat, intputFormat.getParserPosition()));
+				res.add(HiveDDLUtils.toTableOption(STORED_AS_OUTPUT_FORMAT, outputFormat, outputFormat.getParserPosition()));
+			}
+			return res;
+		}
+
+		public static HiveTableStoredAs ofFileFormat(SqlParserPos pos, SqlIdentifier fileFormat) throws ParseException {
+			return new HiveTableStoredAs(pos, fileFormat, null, null);
+		}
+
+		public static HiveTableStoredAs ofInputOutputFormat(SqlParserPos pos, SqlCharStringLiteral intputFormat,
+				SqlCharStringLiteral outputFormat) throws ParseException {
+			return new HiveTableStoredAs(pos, null, intputFormat, outputFormat);
+		}
+	}
+
+	/**
+	 * To represent ROW FORMAT in CREATE TABLE DDL.
+	 */
+	public static class HiveTableRowFormat {
+
+		public static final String SERDE_LIB_CLASS_NAME = "hive.serde.lib.class.name";
+		public static final String SERDE_INFO_PROP_PREFIX = "hive.serde.info.prop.";
+		private static final String FIELD_DELIM = SERDE_INFO_PROP_PREFIX + "field.delim";
```

Comment:
delimiter?

Reply:
These properties are actually SerDe properties defined by Hive. So we can't choose our own name.